api/src/dtos/cards.dto.ts

```diff
@@ -0,0 +1,13 @@
+import { PriceHistoryData } from '@/interfaces/priceHistory.interface';
+import { IsEmail, IsString, IsNotEmpty, MinLength, MaxLength } from 'class-validator';
+
+// Will add a new rating to the list of ratings adjusting the average
+export class CreateRatingDto {
+  public rating: number;
+}
+
+// Will add a new price to PriceHistoryData of a card
+export class CreatePriceHistoryDto {
+  @IsNotEmpty()
+  public historyData: PriceHistoryData;
```

Comment:
To reduce the amount of data you need here, you can probably consider just sending quantity and price, and using BE logic to fill in the month/day/year.

Also, recall that the DTO is not always the same as the interface. If you choose to embed the priceHistoryData, it will also have an _id, for example.

Reply:
ok, this makes more sense now. Thank you.